src/encoded/ingestion_listener.py

```diff
@@ -0,0 +1,670 @@
+             # for practical reasons. -kmp 10-Sep-2020
+             # accept='multipart/form-data',
+             permission='edit')
+@debug_log
+def submit_for_ingestion(context, request):
+    ignored(context)
+
+    check_true(request.content_type == 'multipart/form-data',  # even though we can't declare we accept this
+               "Expected request to have content_type 'multipart/form-data'.", error_class=SubmissionFailure)
+
+    bs_env = beanstalk_env_from_request(request)
+    bundles_bucket = metadata_bundles_bucket(request.registry)
+    datafile = request.POST['datafile']
+    if not isinstance(datafile, cgi.FieldStorage):
+        # e.g., specifically it might be b'' when no file is selected,
+        # but IMPORTANTLY, cgi.FieldStorage has no predefined boolean value,
+        # so we can't just ask to check 'not datafile'. Sigh. -kmp 5-Aug-2020
+        raise UnspecifiedFormParameter('datafile')
+    filename = datafile.filename
+    override_name = request.POST.get('override_name', None)
+    parameters = dict(request.POST)  # Convert to regular dictionary, which is also a copy
+    parameters['datafile'] = filename
+
+    # Other parameters, like validate_only, will ride in on parameters via the manifest on s3
+
+    submission_id, instance = extract_submission_info(request)
+
+    # The three arguments institution, project, and ingestion_type were needed in the old protocol
+    # but are not needed in the new protocol because someone will have set up the IngestionSubmission
+    # object already with the right values. We tolerate them here, but we insist they be consistent (redundant).
+    # Note, too, that we use the 'update=True' option that causes them to be added to our parameters if they are
+    # missing, defaulted from the previous item, so that they will be written to the parameter block stored on S3.
+    # (We could do that differently now, by looking them up dynamically, but rather than risk making a mistake,
+    # I just went with path of least resistance for now.)
+    # -kmp 2-Dec-2020
+
+    if instance.get("institution"):
+        institution = instance['institution']['@id']
+        institution_arg = get_parameter(parameters, "institution", default=institution, update=True)
+        if institution_arg != institution:
+            # If the "institution" argument was passed, which we no longer require, make sure it's consistent.
+            raise SubmissionFailure("'institution' was supplied inconsistently for submit_for_ingestion.")
+
+    if instance.get("project"):
+        project = instance['project']['@id']
+        project_arg = get_parameter(parameters, "project", default=project, update=True)
+        if project_arg != project:
+            # If the "project" argument was passed, which we no longer require, make sure it's consistent.
+            raise SubmissionFailure("'project' was supplied inconsistently for submit_for_ingestion.")
```

Comment:
These should probably go to `project` and `award`, no? @netsettler what do you think?

Reply:
I think you mean `award` and `lab` no (as opposed to `institution` and `project`) no?